test/e2e/tests/test_firewall.py

```diff
@@ -0,0 +1,154 @@
+    replacements["TRAFFIC_TYPE"] = "ALL"
+    replacements["FIREWALL_POLICY_ARN"] = policy_arn
+
+    # Load Network Firewall CR
+    resource_data = load_networkfirewall_resource(
+        "firewall",
+        additional_replacements=replacements,
+    )
+
+    # Create k8s resource
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP,
+        CRD_VERSION,
+        RESOURCE_PLURAL,
+        resource_name,
+        namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+
+    time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+    # Get latest Network Firewall CR
+    cr = k8s.wait_resource_consumed_by_controller(ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(ref)
+
+    yield (ref, cr)
+
+    # Try to delete, if doesn't already exist
+    try:
+        _, deleted = k8s.delete_custom_resource(ref, 3, 10)
+        assert deleted
+    except:
+        pass
+
+@pytest.fixture
+def simple_firewall_policy(request):
+    resource_name = random_suffix_name("fw-pol-ack-test", 24)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["FIREWALL_POLICY_NAME"] = resource_name
+    replacements["STATEFUL_DEFAULT_ACTION"] = "aws:drop_established"
+    replacements["RULE_ORDER"] = "STRICT_ORDER"
+    replacements["STREAM_EXCEPTION_POLICY"] = "DROP"
+    replacements["STATELESS_DEFAULT_ACTION"] = "aws:drop"
+    replacements["STATELESS_FRAG_DEFAULT_ACTION"] = "aws:forward_to_sfe"
+
+    # Load Firewall Policy CR
+    resource_data = load_networkfirewall_resource(
+        "firewall_policy",
+        additional_replacements=replacements,
+    )
+
+    # Create k8s resource
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP,
+        CRD_VERSION,
+        "firewallpolicies",
+        resource_name,
+        namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+
+    time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+    # Get latest Firewall Policy CR
+    cr = k8s.wait_resource_consumed_by_controller(ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(ref)
+
+    yield (ref, cr)
+
+    # Try to delete, if doesn't already exist
+    try:
+        _, deleted = k8s.delete_custom_resource(ref, 3, 10)
+        assert deleted
+    except:
+        pass
+
+
+@service_marker
+@pytest.mark.canary
+class TestNetworkFirewall:
+    def test_create_delete(self, networkfirewall_client, simple_firewall, simple_firewall_policy):
+        simple_firewall_policy
+        (fw, cr) = simple_firewall
+
+        firewall_config = cr["status"]["firewall"]
+        firwall_name = firewall_config["firewallName"]
+
+        networkfirewall_validator = NetworkFirewallValidator(networkfirewall_client)
+        # Check Network Firewall comes to READY state
+        networkfirewall_validator.wait_for_firewall_creation_or_die(firwall_name, "READY", TIMEOUT_SECONDS)
+
+        # Check Network Firewall logging configuration exist in AWS
```

Comment:
Since `hooks.go` is bringing a lot of update code, can we add a test for updating logging configurations?

Reply:
done